### PR TITLE
Use HOSTNAME in Docker build image tag hash

### LINKED ISSUE #24661: Verify flake: verify-generated-protobuf.sh
```
+++ [0422 02:01:46] Creating data container
Error response from daemon: open /var/lib/docker/containers/39f63297738b06376d88f1295c428572710aa560223e3c79d9bcbef891e5da07/config.json: no such file or directory
!!! Error in /go/src/k8s.io/kubernetes/hack/../build/common.sh:595
  '"${docker_cmd[@]}" "$@"' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/../build/common.sh:595 kube::build::run_build_command(...)
  2: /go/src/k8s.io/kubernetes/hack/update-generated-protobuf.sh:57 main(...)
Exiting with status 1
!!! Error in /go/src/k8s.io/kubernetes/hack/after-build/verify-generated-protobuf.sh:44
  '"${KUBE_ROOT}/hack/update-generated-protobuf.sh"' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/after-build/verify-generated-protobuf.sh:44 main(...)
Exiting with status 1
!!! Error in ./hack/../hack/verify-generated-protobuf.sh:26
  '"${KUBE_ROOT}/hack/after-build/verify-generated-protobuf.sh" "$@"' exited with status 1
Call stack:
  1: ./hack/../hack/verify-generated-protobuf.sh:26 main(...)
Exiting with status 1
FAILED   ./hack/../hack/verify-generated-protobuf.sh	3s
```
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24291/kubernetes-pull-test-unit-integration/22388/

cc @kubernetes/sig-testing @wojtek-t

### PR DESCRIPTION
**Pull request by ixdy**
Fixes #24661 by including `$HOSTNAME` when generating the build image tag hash.

When running the verification checks under Docker, the `$KUBE_ROOT` will be identical across builds, so tags will collide unless we add additional uniqueness. By default, the hostname inside a Docker container is its ID, which should be unique enough for us.

I also deleted a misleading error message from the same check.

@kubernetes/sig-testing

### CODE REVIEW

File: hack/after-build/verify-generated-protobuf.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ done
 # We would like to use "sudo" when running on Travis and
 # not use "sudo" when running on Jenkins.
 SUDO="sudo"
-sudo -h > /dev/null || SUDO=""
+sudo -h &> /dev/null || SUDO=""
```
Comment (spxtr) [inline hack/after-build/verify-generated-protobuf.sh]: The comment makes it seem like this is only necessary on Travis. Is it still needed?

Comment (ixdy) [inline hack/after-build/verify-generated-protobuf.sh]: probably not. deleting.

**Comment by ixdy** [comment]
I've also updated `hack/update-generated-protobuf.sh` to use `$REPO_DIR` instead of `$KUBE_ROOT`, since when set, `$REPO_DIR` should be more unique.

File: hack/after-build/verify-generated-protobuf.sh (hunk 1 above)
Comment (ixdy) [inline hack/after-build/verify-generated-protobuf.sh]: actually, we still need this for running the check locally as an unprivileged user. I can clean it up a bit, though.

**Comment by ixdy** [comment]
Reworked the `sudo` code to be clearer.

**Comment by spxtr** [comment]
This may have broken the build. https://console.cloud.google.com/storage/kubernetes-jenkins/logs/kubernetes-build/9650/

**Pull request merged**